Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -401,30 +481,97 @@ void createSeedsForGroupSycl(
       //************************************************//
 
-      const auto maxMemoryAllocation = std::min(
-          edgesComb, globalBufferSize / uint32_t((sizeof(detail::TripletData) +
-                                                  sizeof(detail::SeedData)) *
-                                                 2));
+      // EXPLANATION OF INDEXING (second part)
+      /*
+        For the triplet search, we calculate the upper limit of constructible
+        triplets.
+
+        For this, we multiply the number of compatible bottom and compatible
+        top SPs for each middle SP, and add these together.
+        (nb0*nt0 + nb1*nt1 + ... where nbk is the number of compatible bottom
+        SPs for the kth middle SP, similarly ntb is for tops)
+
+        sumBotTopCombined is a prefix sum array (of length M+1) of the
+        calculated combinations.
 
-      detail::TripletData* deviceCurvImpact =
-          cl::sycl::malloc_device<detail::TripletData>(maxMemoryAllocation,
-                                                       (*q));
+        sumBotTopCombined:
+        ________________________________________________________
+        |     |         |                   |     |  M         | M = number
+        |  0  | nb0*nt0 | nb0*nt0 + nb1*nt1 | ... |  ∑ nbi+nti | of middle
+        |_____|_________|___________________|_____|_i=0________| space points
+
+        We will start kernels and reserve memory for these combinations but
+        only so much we can fit into memory at once.
+
+        We limit our memory usage to globalBufferSize/2, this is currently
+        hard-coded, but it could be configured. Actually, it would be better
+        to use a separate object that manages memory allocations and
+        deallocations and we could ask it to lend us as much memory as it is
+        happy to give.
+
+        For later, let maxMemoryAllocation be maximum allocatable memory for
+        triplet search.
+
+        We start by adding up summing the combinations, until we arrive at a
+        k which for:
+
+        k+1
+         ∑ nbi+nti > maxMemoryAllocation
+        i=0
+        (or k == M).
+
+        So we know, that we need to start our first kernel for the first k
+        middle SPs.
+
+        Inside the triplet search kernel we start with a binary search, to
+        find out which middle SP the thread corresponds to. Note, that
+        sumBotTopCombined is a monotone increasing series of values which
+        allows us to do a binary search on it.
+
+        Inside the triplet search kernel we count the triplets for fixed
+        bottom and middle SP. This is deviceCountTriplets.
+
+        The triplet filter kernel is calculated on threads equal to all
+        possible bottom-middle combinations, which are the sum of found
+        compatible bottom-middle duplets during the duplet search, which is
+        edgedBottom. But not exactly, because we only did the triplet search
```

Comment:
```suggestion
        edgesBottom. But not exactly, because we only did the triplet search
```
I've been reading this paragraph for almost 5 minutes now :D 

maybe just shorten it to: The number of threads for the triplet filter kernel is calculated over all bottom-middle combinations in the current memory allocation

Reply:
I know, sorry. I did write that differently in the docs. I'll change it in the comments too.